### PR TITLE
feat(ticketer): add presence

### PR DESCRIPTION
**Pull request by mezotv**
## Description 
A rich presence addon for ticketer. A ticket bot with a web dashboard for discord.

## Acknowledgements
 * [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
 * [x] I linted the code by running `yarn format`
 * [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

 ## Screenshots
 Proof showing the creation/modification is working as expected

image
image

### CODE REVIEW

**Comment by mezotv** [comment]
Will add that null check tomorrow! Can't think straight right now.

**Comment by DarkVillager** [comment]
The website is down?
image

**Comment by mezotv** [comment]
> The website is down? image

Probably only a temporary issue. Should be up soon!

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/T/Ticketer/presence.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			if (document.querySelector(".server-info > h3")) {
+				title = document.querySelector(".server-info > h3");
+				logo = document.querySelector(".server-info > img");
+				dashboardInfo = document.querySelector(".informations-nav > h2");
+
+				presenceData.details = "Server Settings - Editing:";
+				presenceData.state = title.textContent;
+				presenceData.smallImageKey = logo.getAttribute("src");
+			}
+			if (!dashboardInfo?.textContent) return;
```
Comment (DarkVillager) [inline websites/T/Ticketer/presence.ts:40]: You could just give the text "Browsing the dashboard"?
```suggestion
			if (!dashboardInfo?.textContent) return;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/T/Ticketer/presence.ts (hunk 1 above)
Comment (mezotv) [reply websites/T/Ticketer/presence.ts:40]: Yea that sounds much better! Will change that real quick.

File: websites/T/Ticketer/presence.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		case "/changelog": {
+			presenceData.details = "Viewing the changelog-page";
+			break;
+		}
+		case "/contribute": {
+			presenceData.details = "Viewing the contribution-page";
+			break;
+		}
+		case "/dashboard": {
+			presenceData.details = "Browsing the dashboard";
```
Comment (RisingSunLight42) [inline websites/T/Ticketer/presence.ts:27]: Would be better if those case are in a Record like here : https://github.com/PreMiD/Presences/blob/main/websites/V/Voiranime/presence.ts

Comment (mezotv) [reply websites/T/Ticketer/presence.ts:27]: I kine like the way it currently is easy to read. That just seems like an extra step.

Comment (mezotv) [reply websites/T/Ticketer/presence.ts:27]: I tried adding Records like you but they would require way to much code change and I think its easier if I stick to my simple solution!

**Comment by mezotv** [comment]
Not sure why the first check failed its the same code but I removed the - from the strings.

**Comment by mezotv** [comment]
> rate limit

So what do I do now?

**Comment by Kuriel23** [comment]
> So what do I do now?

Nothing, you don't have to worry about it when you give rate limit.

File: websites/T/Ticketer/metadata.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.7",
+  "service": "Ticketer",
+  "author": {
+    "id": "347077478726238228",
+    "name": "Dominik#5555"
```
Comment (Kuriel23) [inline websites/T/Ticketer/metadata.json]: Remove the discriminator

Comment (mezotv) [reply websites/T/Ticketer/metadata.json]: Done!

**Review by DarkVillager — APPROVED** [review]

**Review by RisingSunLight42 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

**Pull request merged**